kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -339,20 +339,10 @@ class BatchJobSubmission(
 
 object BatchJobSubmission {
   def applicationFailed(applicationStatus: Option[ApplicationInfo]): Boolean = {
-    applicationStatus.map(_.state).exists {
-      case ApplicationState.FAILED => true
-      case ApplicationState.KILLED => true
-      case _ => false
-    }
+    applicationStatus.map(_.state).exists(ApplicationState.isFailed)
   }
 
   def applicationTerminated(applicationStatus: Option[ApplicationInfo]): Boolean = {
-    applicationStatus.map(_.state).exists {
-      case ApplicationState.FAILED => true
-      case ApplicationState.KILLED => true
-      case ApplicationState.FINISHED => true
-      case ApplicationState.NOT_FOUND => true
```

Comment:
Here I remove the `NOT_FOUND`, it should not be in terminate state(might has not been submitted to RM). cc @ulysses-you

Reply:
Seems NOT_FOUND is need for batch, revert it.